mypy.ini

```diff
@@ -9,83 +9,189 @@ mypy_path = stubs
-  synapse/__init__.py,
-  synapse/api,
-  synapse/appservice,
-  synapse/config,
-  synapse/crypto,
-  synapse/event_auth.py,
-  synapse/events,
-  synapse/federation,
-  synapse/groups,
-  synapse/handlers,
-  synapse/http,
-  synapse/logging,
-  synapse/metrics,
-  synapse/module_api,
-  synapse/notifier.py,
-  synapse/push,
-  synapse/replication,
-  synapse/rest,
-  synapse/server.py,
-  synapse/server_notices,
-  synapse/spam_checker_api,
-  synapse/state,
-  synapse/storage/__init__.py,
-  synapse/storage/_base.py,
-  synapse/storage/background_updates.py,
-  synapse/storage/databases/main/appservice.py,
-  synapse/storage/databases/main/client_ips.py,
-  synapse/storage/databases/main/events.py,
-  synapse/storage/databases/main/keys.py,
-  synapse/storage/databases/main/pusher.py,
-  synapse/storage/databases/main/registration.py,
-  synapse/storage/databases/main/relations.py,
-  synapse/storage/databases/main/session.py,
-  synapse/storage/databases/main/stream.py,
-  synapse/storage/databases/main/ui_auth.py,
-  synapse/storage/databases/state,
-  synapse/storage/database.py,
-  synapse/storage/engines,
-  synapse/storage/keys.py,
-  synapse/storage/persist_events.py,
-  synapse/storage/prepare_database.py,
-  synapse/storage/purge_events.py,
-  synapse/storage/push_rule.py,
-  synapse/storage/relations.py,
-  synapse/storage/roommember.py,
-  synapse/storage/state.py,
-  synapse/storage/types.py,
-  synapse/storage/util,
-  synapse/streams,
-  synapse/types.py,
-  synapse/util,
-  synapse/visibility.py,
-  tests/replication,
-  tests/test_event_auth.py,
-  tests/test_utils,
-  tests/handlers/test_password_providers.py,
-  tests/handlers/test_room.py,
-  tests/handlers/test_room_summary.py,
-  tests/handlers/test_send_email.py,
-  tests/handlers/test_sync.py,
-  tests/handlers/test_user_directory.py,
-  tests/rest/client/test_login.py,
-  tests/rest/client/test_auth.py,
-  tests/rest/client/test_relations.py,
-  tests/rest/media/v1/test_filepath.py,
-  tests/rest/media/v1/test_oembed.py,
-  tests/storage/test_state.py,
-  tests/storage/test_user_directory.py,
-  tests/util/test_itertools.py,
-  tests/util/test_stream_change_cache.py
+  # Discover all .py[i] files in the repo
+  .,
+
+  # Also check scripts that don't end in .py
+  scripts/export_signing_key,
+  scripts/generate_config,
+  scripts/generate_log_config,
+  scripts/hash_password,
+  scripts/register_new_matrix_user,
+ #scripts/synapse_port_db,
+  scripts/synapse_review_recent_signups,
+ #scripts/update_synapse_database,
+  scripts-dev/build_debian_packages,
+  scripts-dev/sign_json
+
+# Note: Better exclusion syntax in mypy > 0.910
+# See https://github.com/python/mypy/pull/11329
+exclude = (?x)
+  ^(contrib/cmdclient/console.py
+   |contrib/cmdclient/http.py
+   |contrib/experiments/cursesio.py
+   |contrib/experiments/test_messaging.py
+   |contrib/graph/graph.py
+   |contrib/graph/graph2.py
+   |contrib/graph/graph3.py
+   |contrib/jitsimeetbridge/jitsimeetbridge.py
```

Comment:
I guess I'm not *entirely* convinced we need to enforce mypy for `contrib` stuff, but :man_shrugging:

Reply:
Yeah, me either. Heck, I'm not even sure we should be shipping `contrib/` in this repo at all 😉 

I'll switch this to a blanket exclusion of `|contrib/.*`